jvm-packages/xgboost4j-spark/src/main/scala/ml/dmlc/xgboost4j/scala/spark/XGBoost.scala

```diff
@@ -130,6 +130,9 @@ object XGBoost extends Serializable {
               trainingSet.setGroup(xgBoostConfMap("groupData").asInstanceOf[Seq[Seq[Int]]](
                   TaskContext.getPartitionId()).toArray)
             }
+            if (params.contains("base_margin")) {
+              trainingSet.setBaseMargin(params("base_margin").asInstanceOf[Array[Float]])
```

Comment:
I can rewrite it in a way that it also supports `Array[Array[Float]]` and fails if the parameter is of different type. What do you think?